code/src/components/products-container/ProductsContainer.tsx

```diff
@@ -21,8 +24,14 @@ const ProductsContainer = ({
   loadingItemsCount = 5,
   errorMessage = "errors.somethingWentWrong"
 }: ProductsContainerProps) => {
+  const user = useUserDetailsSelector();
+
+  const [addToCart] = useAddToCartMutation();
```

Comment:
How do we handle loading state?

Reply:
This is used when user clicks add to cart button inside of ProductCard (demonstrated below). I think loading indicator is redundant there. In case of successful adding to the cart, drawer with new item opens, so user knows that it was added. Otherwise snackbar is shown.

![image](https://github.com/idx-academy/spa-orders/assets/62070431/33f6e2d1-73e9-435e-b720-1a5c7ef7ba52)